modules/genomic_browser/php/endpoints/filemanager.class.inc

```diff
@@ -0,0 +1,352 @@
+               );
+    }
+
+    /**
+     * Provide an array of distinct value from a column in
+     * the specified table. The first element has a null key and
+     * its value is 'Any'.
+     *
+     * @param string $table  The tables name
+     * @param string $column The column name
+     *
+     * @return array An array where value are keys as well as values
+     */
+    function getDistinctValues( $table, $column)
+    {
+        $DB      = \Database::singleton();
+        $results = $DB->pselect(
+            "SELECT DISTINCT $column FROM $table ",
+            array()
+        );
+        return array_reduce(
+            $results,
+            function ($carry, $item) {
+                $value = array_values($item)[0];
+                if (!empty($value)) {
+                    $carry[str_replace(' ', '_', $value)] = $value;
+                }
+                return $carry;
+            },
+            array(null => 'Any')
+        );
+    }
+
+    /**
+     * Processes the values & saves to database and return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse POST request body.
+        $values = $request->getParsedBody();
+
+        // User permission verification.
+        $values = $this->_verifyPermissions($request, $values);
+
+        if (isset($values['error'])) {
+            switch ($values['error']) {
+            case 'BadRequest':
+                return new \LORIS\Http\Response\BadRequest(
+                    'Failed to retrieve data for create timepoint.'
+                );
+            case 'Forbidden':
+                return new \LORIS\Http\Response\Forbidden(
+                    'Forbidden'
+                );
+            default:
+                break;
+            }
+        }
+
+        // Genomic File for uploading task.
+        $genomicFile = new \LORIS\genomic_browser\endpoints\uploading\Genomicfile();
+
+        // handles upload process - returns status of file upload.
+        $response = $genomicFile->setupUploading(
+            $request->getAttribute('user'), // user
+            $_FILES, // file data
+            $values // post request values.
+        );
+
+        // Send file upload status.
+        return new \LORIS\Http\Response\JsonResponse(
+            $response
+        );
+    }
+
+    /**
+     * Permission verification.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     * @param array                  $values  The form values.
+     *
+     * @return array
+     */
+    private function _verifyPermissions(
+        ServerRequestInterface $request,
+        array $values
+    ) : array {
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('genomic_browser_view_site')
```

Comment:
```suggestion
        if (!$user->hasPermission('genomic_browser_view_site')
```

As an FYI, you can use `hasAnyPermissions` and pass an array of permissions if you want.

Reply:
Sorry this should've said `hasAnyPermission`. @maltheism. No `s`